File: src/game/entity/enemies/npcs/Eagle.java
package game.entity.enemies.npcs;

import game.utils.Paths;

public class Eagle extends FlyingEnemy {
    @Override
    protected String getBasePath() {
        return Paths.getEnemiesFolder() + "/eagle";
    }

    public String[] getCharacterOrientedImages() {
        return new String[]{
                String.format("%s/eagle_%s_0.png", getBasePath(), imageDirection.toString().toLowerCase()),
                String.format("%s/eagle_%s_1.png", getBasePath(), imageDirection.toString().toLowerCase()),
                String.format("%s/eagle_%s_2.png", getBasePath(), imageDirection.toString().toLowerCase()),
        };
    }
}

File: src/game/entity/enemies/npcs/Eagle.kt
package game.entity.enemies.npcs

import game.entity.EntityTypes
import game.entity.models.Coordinates
import game.utils.Paths.enemiesFolder

class Eagle : FlyingEnemy {
    constructor(id: Long) : super(id)
    constructor(coordinates: Coordinates?) : super(coordinates)
    constructor() : super()

    override val entitiesAssetsPath: String get() ="$enemiesFolder/eagle"

    override fun getCharacterOrientedImages(): Array<String> = Array(3) { index ->
        "$entitiesAssetsPath/eagle_${imageDirection.toString().lowercase()}_$index.png"
    }

    override val type: EntityTypes
        get() = EntityTypes.Eagle
}